src/main/java/com/couchbase/lite/SavedRevision.java

```diff
@@ -201,7 +202,33 @@ public long getSequence() {
 
     }
 
+    @SuppressWarnings("unchecked")
+    private static <K,V> Map<K,V> nestedUnmodifiableMap(Map<? extends K, ? extends V> m) {
+        // Don't make a copy of m until we have to modify it.
+        Map<K,V> copy = null;
+
+        for (Map.Entry<? extends K,? extends V> entry : m.entrySet()) {
+            if (entry.getValue() instanceof Map) {
+                if (copy == null) {
+                    copy = new HashMap<K,V>(m);
+                }
+                Map newValue = nestedUnmodifiableMap((Map) entry.getValue());
+                try {
+                    copy.put(entry.getKey(), (V) newValue);
+                } catch (ClassCastException e) {
```

Comment:
Due to type erasure of Java Generics, this try catch is not needed and won't prevent an Object that is not assignable from V to be added to the new map.